### PR TITLE
[Merged by Bors] - feat: port Data.Seq.WSeq

### PR DESCRIPTION
**Pull request by Komyyy**
This PR also make `Option.recOn` computable.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by eric-wieser** [comment]
Please run `git rebase -i origin/master`, remove the commits that are not part of this port, then `git push origin --force-with-lease`

File: Mathlib/Data/Seq/WSeq.lean [hunk 1]
```diff
@@ -0,0 +1,1826 @@
...
+        assumption
+      apply Computation.LiftRel.imp _ _ _ h
+      intro a b
+      apply LiftRelO.imp_right
+      intro s t
+      apply Or.inl⟩⟩
+#align stream.wseq.lift_rel_destruct_iff Stream'.WSeq.liftRel_destruct_iff
+
+-- Porting note: To avoid ambiguous notation, `~` became `~ʷ`.
+-- mathport name: equiv
```
Comment (semorrison) [inline Mathlib/Data/Seq/WSeq.lean:526]:
```suggestion
```

File: Mathlib/Data/Seq/WSeq.lean [hunk 2]
```diff
@@ -0,0 +1,1826 @@
...
+theorem Equiv.equivalence : Equivalence (@Equiv α) :=
+  ⟨@Equiv.refl _, @Equiv.symm _, @Equiv.trans _⟩
+#align stream.wseq.equiv.equivalence Stream'.WSeq.Equiv.equivalence
+
+open Computation
+
+-- Porting note: In Lean4, `return` is a built-in notation, so this command doesn't work,
+--                         so, `return`s are replaced with `Computation.pure`.
+-- local notation "return" => Computation.pure
+
```
Comment (semorrison) [inline Mathlib/Data/Seq/WSeq.lean:629]:
```suggestion
```
I think it's okay to just omit this.

**Comment by semorrison** [comment]
@Komyyy, I pushed a change that removes (pending CI!!) the `simpNF` attribute. If this looks okay and CI passes, could you merge. Otherwise please mark this back as `awaiting-review`.

bors d+

**Pull request closed without merging**